src/Compilers/Core/MSBuildTask/Csc.cs

```diff
@@ -535,6 +535,13 @@ private bool InitializeHostCompiler(ICscHostObject cscHostObject)
                     CheckHostObjectSupport(param = nameof(AdditionalFiles), analyzerHostObject.SetAdditionalFiles(AdditionalFiles));
                 }
 
+                // For host objects which support it, set the analyzer config files and potential config files.
+                if (cscHostObject is IAnalyzerConfigFilesHostObject analyzerConfigFilesHostObject)
+                {
+                    CheckHostObjectSupport(param = nameof(AnalyzerConfigFiles), analyzerConfigFilesHostObject.SetAnalyzerConfigFiles(AnalyzerConfigFiles));
```

Comment:
The string passed through `ICompilerOptionsHostObject` is basically opaque to the project system. Since we need to set up file watchers and incorporate this info into the fast up-to-date check we need to split them out.